apps/webapp/src/state/swap.ts

```diff
@@ -1,103 +1,105 @@
-import { assets } from '@penumbra-zone/constants';
 import { AllSlices, SliceCreator } from './index.ts';
-import { Asset, AssetId } from '@penumbra-zone/types';
-
-export enum SwapInputs {
-  PAY = 'pay',
-  RECEIVE = 'receive',
-}
-
-export interface SwapValidationFields {
-  pay: boolean;
-}
-
-export interface SwapAssetInfo {
-  amount: string;
-  asset: Asset;
-  balance?: number;
-}
+import { Selection } from './types.ts';
+import {
+  errorTxToast,
+  loadingTxToast,
+  successTxToast,
+} from '../components/shared/toast-content.tsx';
+import { toBaseUnit } from '@penumbra-zone/types';
+import { toast } from '@penumbra-zone/ui/components/ui/use-toast.ts';
+import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
+import { AddressIndex } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1alpha1/keys_pb';
+import { getEphemeralAddress } from '../fetchers/address.ts';
+import BigNumber from 'bignumber.js';
+import { planWitnessBuildBroadcast } from './helpers.ts';
+import { Asset } from '@penumbra-zone/constants/assets.ts';
 
 export interface SwapSlice {
-  pay: SwapAssetInfo;
-  receive: SwapAssetInfo;
-  validationErrors: SwapValidationFields;
-  setAmount: (type: SwapInputs) => (amount: string) => void;
-  setAsset: (type: SwapInputs) => (asset: AssetId) => void;
-  replaceAsset: () => void;
-  setAssetBalance: (amount: number) => void;
+  assetIn: Selection | undefined;
+  setAssetIn: (selection: Selection) => void;
+  amount: string;
```

Comment:
Shouldn't this be a `number`? (and get converted in the `InputToken`'s [`onChange` handler](https://github.com/penumbra-zone/web/pull/419/files#diff-0a9c5fc7496ea1a47f7fef2e351c40f80cdc243873cbf333c5412ddac5ee1f81R30-R33))?